src/handlers/swap.ts

```diff
@@ -397,10 +401,29 @@ export const estimateCrosschainSwapGasLimit = async ({
       provider,
       SWAP_GAS_PADDING
     );
+    const routeGasLimit = getCrosschainSwapDefaultGasLimit(tradeDetails);
+    const rainbowDefaultGasLimit = getDefaultGasLimitForTrade(
+      tradeDetails,
+      chainId
+    );
 
-    return gasLimit || getCrosschainSwapDefaultGasLimit(tradeDetails);
+    let fallbackGasLimit: BigNumberish = rainbowDefaultGasLimit;
+    if (routeGasLimit && lessThan(rainbowDefaultGasLimit, routeGasLimit)) {
+      fallbackGasLimit = routeGasLimit;
+    }
+    return gasLimit || fallbackGasLimit;
   } catch (error) {
-    return getCrosschainSwapDefaultGasLimit(tradeDetails);
+    const routeGasLimit = getCrosschainSwapDefaultGasLimit(tradeDetails);
+    const rainbowDefaultGasLimit = getDefaultGasLimitForTrade(
+      tradeDetails,
+      chainId
+    );
+
+    let fallbackGasLimit: BigNumberish = rainbowDefaultGasLimit;
+    if (routeGasLimit && lessThan(rainbowDefaultGasLimit, routeGasLimit)) {
+      fallbackGasLimit = routeGasLimit;
+    }
```

Comment:
just a personal preference i'd do

```
const fallbackGasLimit = lessThan(rainbowDefaultGasLimit, routeGasLimit) ? routeGasLimit : rainbowDefaultGasLimit
```

same for the changes above, but looking great